ml_genn/model.py

```diff
@@ -434,35 +434,57 @@ def convert_tf_model(tf_model, converter=Simple(),
                 # === Dense Layers ===
                 if isinstance(tf_layer, tf.keras.layers.Dense):
                     name = tf_layer.name
-                    weights = []
                     sources = []
                     synapses = []
+                    weights = []
 
                     # create layer
                     mlg_layer = Layer(name=name, neurons=converter.create_neurons(
                         tf_layer, pre_compile_output))
 
                     # create synapses
                     for tf_in_layer in tf_in_layers:
-                        weights.append(tf_layer.get_weights()[0])
-                        sources.append(mlg_layer_lookup[tf_in_layer])
 
-                        if isinstance(tf_in_layer, tf.keras.layers.AveragePooling2D):
-                            synapses.append(AvePool2DDenseSynapses(
-                                units=tf_layer.units,
-                                pool_size=tf_in_layer.pool_size,
-                                pool_strides=tf_in_layer.strides,
-                                pool_padding=tf_in_layer.padding,
-                                connectivity_type=connectivity_type))
-
-                        elif isinstance(tf_in_layer, tf.keras.layers.GlobalAveragePooling2D):
-                            synapses.append(AvePool2DDenseSynapses(
-                                units=tf_layer.units,
-                                pool_size=tf_in_layer.input_shape[1:3],
-                                connectivity_type=connectivity_type))
+                        if isinstance(tf_in_layer, (
+                                tf.keras.layers.AveragePooling2D,
+                                tf.keras.layers.GlobalAveragePooling2D)):
+
+                            # traverse ignored layers to find more pool layer inputs
+                            pool_in_layers = set(tf_in_layers_all[tf_in_layer])
+                            final_pool_in_layers = set()
+                            while pool_in_layers:
+                                pool_in_layer = pool_in_layers.pop()
+                                if isinstance(pool_in_layer, ignored_tf_layers):
+                                    pool_in_layers.update(tf_in_layers_all[pool_in_layer])
+                                else:
+                                    final_pool_in_layers.add(pool_in_layer)
+                            pool_in_layers = final_pool_in_layers
```

Comment:
This logic is duplicated in two other places ([here](https://github.com/genn-team/ml_genn/pull/33/files#diff-1c228c3438adda0d7b0fef132e5b9a1e9b281ef7b7d70f810e08ad4477e8126eR421-R428) and [here](https://github.com/genn-team/ml_genn/pull/33/files#diff-1c228c3438adda0d7b0fef132e5b9a1e9b281ef7b7d70f810e08ad4477e8126eR515-R522)). Maybe this could be moved out into a helper function to aid the readability of this

Reply:
Done.